gapic-generator/test/gapic/mixins/mixins_test.rb

```diff
@@ -45,6 +45,22 @@ def test_mixin_service_address_checker
     refute Gapic::Model::Mixins.mixin_service_address? "testing.mixins.ServiceWithLoc"
   end
 
+  def test_mixin_message_field_address_checker
+    # sic. no `s` at the end, this is a message named `Location`, not service named `Locations`
+    assert Gapic::Model::Mixins.mixin_message_field_address? "google.cloud.location.Location"
+    assert Gapic::Model::Mixins.mixin_message_field_address? "google.cloud.location.Location.metadata"
+
+    refute Gapic::Model::Mixins.mixin_message_field_address? "google.cloud.location.Location",
+                                                             gem_name: "google-cloud-location"
+    refute Gapic::Model::Mixins.mixin_message_field_address? "google.cloud.location.Location.metadata",
+                                                             gem_name: "google-cloud-location"
```

Comment:
Missing test case: gem_name provided but does not match the given address; expected result is true for a mixin address.

Reply:
done